common/src/main/cpp/extensions/openxr_fb_render_model.cpp

```diff
@@ -0,0 +1,131 @@
+#include <godot_cpp/variant/utility_functions.hpp>
+
+using namespace godot;
+
+void OpenXRFbRenderModel::set_render_model_name(Model p_model) {
+	render_model_name = p_model;
+	load_render_model();
+}
+
+OpenXRFbRenderModel::Model OpenXRFbRenderModel::get_render_model_name() {
+	return render_model_name;
+}
+
+bool OpenXRFbRenderModel::has_render_model() {
+	return render_model != nullptr;
+}
+
+void OpenXRFbRenderModel::load_render_model() {
+	if (render_model != nullptr) {
+		render_model->queue_free();
+		render_model = nullptr;
+	}
+
+	String render_model_path;
+	switch (render_model_name) {
+		case MODEL_CONTROLLER_LEFT: {
+			render_model_path = "/model_fb/controller/left";
+		} break;
+
+		case MODEL_CONTROLLER_RIGHT: {
+			render_model_path = "/model_fb/controller/right";
+		} break;
+
+		default: {
+			render_model_path = "";
+		}
+	}
+
+	if (render_model_path.is_empty()) {
+		return;
+	}
+
+	PackedByteArray render_model_buffer = OpenXRFbRenderModelExtensionWrapper::get_singleton()->get_buffer(render_model_path);
+	if (render_model_buffer.is_empty()) {
+		UtilityFunctions::print_verbose("Failed to load render model buffer from path [", render_model_path, "] in OpenXRFbRenderModel node");
+		return;
+	}
+
+	Ref<GLTFDocument> gltf_document;
+	gltf_document.instantiate();
+	Ref<GLTFState> gltf_state;
+	gltf_state.instantiate();
+
+	Error err = gltf_document->append_from_buffer(render_model_buffer, "", gltf_state);
+	if (err != OK) {
+		UtilityFunctions::print("Failed to instance render model in OpenXRFbRenderModel node");
+		return;
+	}
+
+	render_model = Object::cast_to<Node3D>(gltf_document->generate_scene(gltf_state));
+	if (render_model) {
+		add_child(render_model);
+		emit_signal("render_model_loaded");
+	}
+}
+
+Node3D *OpenXRFbRenderModel::get_render_model() {
+	return render_model;
+}
+
+void OpenXRFbRenderModel::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_POSTINITIALIZE: {
+			Ref<OpenXRInterface> openxr_interface = XRServer::get_singleton()->find_interface("OpenXR");
+			if (openxr_interface.is_valid()) {
+				openxr_interface->connect("session_begun", callable_mp(this, &OpenXRFbRenderModel::load_render_model));
+			}
+		} break;
+	}
+}
+
+void OpenXRFbRenderModel::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("has_render_model"), &OpenXRFbRenderModel::has_render_model);
+	ClassDB::bind_method(D_METHOD("get_render_model"), &OpenXRFbRenderModel::get_render_model);
+	ClassDB::bind_method(D_METHOD("set_render_model_name", "render_model_name"), &OpenXRFbRenderModel::set_render_model_name);
+	ClassDB::bind_method(D_METHOD("get_render_model_name"), &OpenXRFbRenderModel::get_render_model_name);
+
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "render_model_name", PROPERTY_HINT_ENUM, "Left Controller,Right Controller"), "set_render_model_name", "get_render_model_name");
+
+	BIND_ENUM_CONSTANT(MODEL_CONTROLLER_LEFT);
+	BIND_ENUM_CONSTANT(MODEL_CONTROLLER_RIGHT);
+
+	ADD_SIGNAL(MethodInfo("render_model_loaded"));
```

Comment:
Let's make the signal name more explicit and use `openxr_fb_render_model_loaded`.